api/src/main/resources/workbench-api.yaml

```diff
@@ -7280,3 +7329,95 @@ definitions:
         format: int64
       conceptName:
         type: string
+  AuditLogEntriesResponse:
+    type: object
+    description: >
+      Represents the results of a query into the audit dataset in BigQery without
+      filtering or transformation. This response type is a stopgap until we develop
+      high-level queries with a better semantic mapping to the concepts and rules
+      governing them, which is why a freeform queryMetadata field is tolerable for
+      now.
+    required:
+    - logEntries
+    - queryMetadata
+    properties:
+      logEntries:
+        type: array
+        items:
+          "$ref": "#/definitions/AuditLogEntry"
+      queryMetadata:
```

Comment:
I'm generally not a fan of unspecified objects / fields in APIs. There really isn't too much additional cost in specifying field names and types, and IMO there's a huge benefit in API specificity.

There's also no other use of "additionalProperties" in this file, which suggests an argument by consistency, that we have a strong pattern to fully specify our API response objects.

Reply:
I generally agree. There's a historical problem here, because the audit system uses the DB id and the API does not.

The reason to do it here is to allow reusing the response object with several different kinds of queries. For example, the user query endpoint might take in a user ID, but the output is in terms of the database identifier.

Now, in the special case of this query, the DB ID is easy to extract from the result set, since it only shows up in one column. In the case of hte User query, I'm probably going to include both rows where the user is the agent as well as the target, so it's not obvious which ID should be used.

The client will likely be interested in distinguishing which column is describing the user in question.

Now, why not just have a `databaseId` field for both the user and the workspace (and others)? That gets you a little way down the road, but then you can easily have queries with more than one input (in principle).

The best solution not involving an arbitrary map is separate response types for each query type that look very similar but all contain a list of rows. That's cheap enough I suppose.

The balancing factor was that this was all going to be temporary until we built more high-level "storytelling" APIs for different types of things for different use cases. But that colud be a while.

Reply:
I think I like the solution in your second to last paragraph best – separate out the user-and-workspace specific response types, but use a common AuditLogEntries data definition for each.